lib/system.nim

```diff
@@ -1871,6 +1871,21 @@ proc `$` *[Enum: enum](x: Enum): string {.magic: "EnumToStr", noSideEffect.}
   ## a ``$`` operator for a concrete enumeration is provided, this is
   ## used instead. (In other words: *Overwriting* is possible.)
 
+proc newString*[N](data: array[N, char]): string {.noSideEffect.} =
```

Comment:
Well I don't generally have a problem with  ``toCstring``, but I also don't see a direct necessity to do so. It's a new name that needs to be introduced and remembered, and the cast works fine in my opinion.